### PR TITLE
Remove submission local data (EXPOSUREAPP-5379)

### PR DESCRIPTION
**Pull request by kolyaopahle**
### Description

This PR removes all submission related values from local data and into submission settings
it also does some refactoring to how the data is handled

### Steps to reproduce
1. install the App
2. Check if submission works as expected on a new install

### CODE REVIEW

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/worker/BackgroundWorkScheduler.kt [hunk 1]
```diff
@@ -17,7 +18,11 @@ import java.util.concurrent.ExecutionException
  * @see BackgroundConstants
  * @see BackgroundWorkHelper
  */
-object BackgroundWorkScheduler {
+object BackgroundWorkScheduler : BackgroundWorkSchedulerBase() {
+
+    fun init(component: ApplicationComponent) {
```
Comment (d4rken) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/worker/BackgroundWorkScheduler.kt:23]: Where is this called?

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/worker/BackgroundWorkSchedulerBase.kt [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package de.rki.coronawarnapp.worker
+
+import de.rki.coronawarnapp.submission.SubmissionSettings
+import javax.inject.Inject
+
+@Suppress("UnnecessaryAbstractClass")
+abstract class BackgroundWorkSchedulerBase {
+    @Inject internal lateinit var submissionSettings: dagger.Lazy<SubmissionSettings>
```
Comment (d4rken) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/worker/BackgroundWorkSchedulerBase.kt:8]: Does this need to be `lazy`?

File: Corona-Warn-App/src/test/java/de/rki/coronawarnapp/storage/SubmissionRepositoryTest.kt [hunk 3]
```diff
@@ -60,6 +60,9 @@ class SubmissionRepositoryTest : BaseTest() {
     private val testResult = TestResult.PENDING
     private val registrationData = SubmissionService.RegistrationData(registrationToken, testResult)
 
+    private val registrationTokenPreference = mockFlowPreference<String?>(null)
+    private val resultReceivedTimeStamp = Instant.ofEpochMilli(101010101)
```
Comment (d4rken) [inline Corona-Warn-App/src/test/java/de/rki/coronawarnapp/storage/SubmissionRepositoryTest.kt:64]: Any specific reason the test value changed from `1` to this?

**Review by d4rken — COMMENTED** [review]
Good cleanup :ok_hand:

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/worker/BackgroundWorkScheduler.kt (hunk 1 above)
Comment (kolyaopahle) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/worker/BackgroundWorkScheduler.kt:23]: https://github.com/corona-warn-app/cwa-app-android/pull/2505/files#diff-fdd468b18ccf30dc7a16638bc7f0032deedf65a479ca506ee7406a6380f9cc6fR78 in onCreate of CoronaWarnApplication

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/worker/BackgroundWorkSchedulerBase.kt (hunk 2 above)
Comment (kolyaopahle) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/worker/BackgroundWorkSchedulerBase.kt:8]: probably not

Comment (kolyaopahle) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/worker/BackgroundWorkSchedulerBase.kt:8]: i stole it from the other base impl where it was also lazy

File: Corona-Warn-App/src/test/java/de/rki/coronawarnapp/storage/SubmissionRepositoryTest.kt (hunk 3 above)
Comment (kolyaopahle) [reply Corona-Warn-App/src/test/java/de/rki/coronawarnapp/storage/SubmissionRepositoryTest.kt:64]: the old version of the initialTestResultReceivedTimestamp was a long that was considered as unset if it was 0, we now store an Instant? that is unset if it is null, the test only checks for null / != null and does not care about the specific value, i decided to have a more specific time than just 1 in the Instant to make it more obvious that its not just a one off error in some places

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/submission/SubmissionSettings.kt [hunk 4]
```diff
@@ -35,25 +36,54 @@ class SubmissionSettings @Inject constructor(
         context.getSharedPreferences("submission_localdata", Context.MODE_PRIVATE)
     }
 
+    val registrationToken = prefs.createFlowPreference<String?>(
+        key = TEST_REGISTRATION_TOKEN,
+        defaultValue = null
+    )
+
+    var initialTestResultReceivedAt: Instant?
+        get() = prefs.getLong(TEST_RESULT_RECEIVED_AT, 0L).let {
```
Comment (BMItter) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/submission/SubmissionSettings.kt:45]: to keep your high code quality, dont repeat yourself :octocat: 
..get() = ....

**Comment by BMItter** [comment]
👌

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/submission/SubmissionSettings.kt [hunk 5]
```diff
@@ -85,4 +115,19 @@ class SubmissionSettings @Inject constructor(
     fun clear() {
         prefs.clearAndNotify()
     }
+
+    companion object {
```
[the remark this answers is not in the document]
Comment (kolyaopahle) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/submission/SubmissionSettings.kt:119]: i'm following the pattern that was used in other Settings classes

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/submission/SubmissionSettings.kt (hunk 4 above)
Comment (kolyaopahle) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/submission/SubmissionSettings.kt]: made the code a bit more DRY

File: Corona-Warn-App/src/test/java/de/rki/coronawarnapp/datadonation/analytics/modules/registeredtest/TestResultDonorTest.kt [hunk 6]
```diff
@@ -47,13 +48,14 @@ class TestResultDonorTest : BaseTest() {
         every { riskLevelSettings.lastChangeCheckedRiskLevelTimestamp } returns Instant.now()
         every { testResultDonorSettings.riskLevelAtTestRegistration } returns
             mockFlowPreference(PpaData.PPARiskLevel.RISK_LEVEL_LOW)
-        every { LocalData.initialTestResultReceivedTimestamp() } returns System.currentTimeMillis()
+        every { submissionSettings.initialTestResultReceivedAt } returns Instant.now()
```
[the remark this answers is not in the document]
Comment (kolyaopahle) [reply Corona-Warn-App/src/test/java/de/rki/coronawarnapp/datadonation/analytics/modules/registeredtest/TestResultDonorTest.kt]: done 👯

**Comment by LukasLechnerDev** [comment]
I see the onboarding every time I start the app 🤔

**Comment by kolyaopahle** [comment]
> I see the onboarding every time I start the app 🤔

as far as i can tell no code in this PR touches the onboarding logic

**Comment by LukasLechnerDev** [comment]
Submission is working on my device. Tested different QR codes and TeleTan. 👍

**Comment by jurajkusnier** [comment]
> I see the onboarding every time I start the app 🤔

The migration method deletes encrypted preferences on every app start but the file is also recreated since LocalData is not fully removed and some part of the app needs it. It will be fixed in another PR.

**Pull request merged**